spec/lib/appsignal/hooks/gvl_spec.rb

```diff
@@ -0,0 +1,128 @@
+describe Appsignal::Hooks::GvlHook do
+  before(:context) do
+    Appsignal.config = project_fixture_config
+  end
+
+  def allow_gvltools_require
+    allow_any_instance_of(described_class).to receive(:require).with("gvltools").and_return(true)
+  end
+
+  context "without GVLTools" do
+    describe "#dependencies_present?" do
+      context "if requiring gvltools fails" do
+        it "is false" do
+          expect(described_class.new.dependencies_present?).to be_falsy
+        end
+      end
+
+      it "is false" do
+        allow_gvltools_require
+        expect(described_class.new.dependencies_present?).to be_falsy
+      end
+    end
+  end
+
+  context "with old versions of GVLTools" do
+    before(:context) do
+      module GVLTools
+        VERSION = "0.1.0".freeze
+      end
+    end
+
+    after(:context) { Object.send(:remove_const, :GVLTools) }
+
+    describe "#dependencies_present?" do
+      it "is false" do
+        allow_gvltools_require
+        expect(described_class.new.dependencies_present?).to be_falsy
+      end
+    end
+  end
+
+  context "with new versions of GVLTools" do
+    before(:context) do
+      module GVLTools
+        VERSION = "0.2.0".freeze
+
+        module GlobalTimer
+          def self.enable
+          end
+        end
+
+        module WaitingThreads
+          def self.enable
+          end
+        end
+      end
+    end
+
+    after(:context) { Object.send(:remove_const, :GVLTools) }
+
+    before(:each) { allow_gvltools_require }
+
+    describe "#dependencies_present?" do
+      if DependencyHelper.ruby_3_2_or_newer?
+        it "is true" do
+          expect(described_class.new.dependencies_present?).to be_truthy
+        end
+      else
+        it "is false" do
+          expect(described_class.new.dependencies_present?).to be_falsy
+        end
+      end
+    end
+
+    if DependencyHelper.ruby_3_2_or_newer?
+      describe "Appsignal::Hooks.load_hooks" do
+        # After installing a hook once, it is marked as already installed,
+        # and subsequent calls to `load_hooks` silently do nothing.
+        # Because of this, only one of the tests for the installation uses
+        # `load_hooks`, while the rest call the `install` method directly.
+
+        it "should be added to minutely probes" do
```

Comment:
```suggestion
        it "is added to minutely probes" do
```
Using "should" in the spec names is an old standard, back from when RSpec assertions looked like `object.should eq(:something)`. Our old tests may still use this format, but new specs shouldn't.